x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Configuration.java

```diff
@@ -68,8 +78,11 @@ public EsqlConfiguration(
         assert tables != null;
     }
 
-    public EsqlConfiguration(BlockStreamInput in) throws IOException {
-        super(in.readZoneId(), Instant.ofEpochSecond(in.readVLong(), in.readVInt()), in.readOptionalString(), in.readOptionalString());
+    public Configuration(BlockStreamInput in) throws IOException {
+        this.zoneId = in.readZoneId();
+        this.now = Instant.ofEpochSecond(in.readVLong(), in.readVInt()).atZone(zoneId);
+        this.username = in.readOptionalString();
+        this.clusterName = in.readOptionalString();
```

Comment:
I _think_ clustername is always present, we are taking it from the clusterService. So maybe not optional?

Reply:
I can leave a TODO about it but I don't want to mix the serialization change with the refactoring.